src/parachain/collateral.ts

```diff
@@ -59,15 +60,25 @@ export class DefaultCollateralAPI implements CollateralAPI {
         return new Big(planckToDOT(account.free.toString()));
     }
 
-    async transfer(address: string, amount: string | number): Promise<void> {
-        if (!this.account) {
-            return Promise.reject(ACCOUNT_NOT_SET_ERROR_MESSAGE);
+    async subscribeToBalance(account: string, callback: (account: string, balance: Big) => void): Promise<() => void> {
+        try {
+            const accountId = this.api.createType("AccountId", account);
+            const unsubscribe = await this.api.query.dot.account(accountId, (balance) => {
+                callback(account, new Big(planckToDOT(balance.free.toString())));
+            });
+            return unsubscribe;
+        } catch (error) {
+            console.log(`Error during collateral balance subscription callback: ${error}`);
         }
-        const transferTx = this.api.tx.dot.transfer(address, amount);
-        await this.transaction.sendLogged(transferTx, this.account, this.api.events.dot.Transfer);
+        // as a fallback, return an empty void function
+        return () => {
+            return;
+        };
     }
 
-    setAccount(account: AddressOrPair): void {
-        this.account = account;
+    async transfer(address: string, amount: Big): Promise<void> {
+        const amountSmallDenomination = this.api.createType("Balance", dotToPlanck(amount.toString()));
```

Comment:
I guess for a future PR, we should change this to take `Big` and return a `BN`

Reply:
Would the `BN` represent the remaining balance? In this case I think `Big` would be appropriate (it may be reused immediately to do another transfer)

Reply:
I was thinking `BN` just because the smallest denomination can't have any decimal places. So the transfer can on the chain side take at the bare minimum `1` (not considering dust values).

Reply:
Oh, so `transfer` would return the Planck amount that actually got submitted?
If an amount smaller than `1` is passed, a runtime error is thrown by the `Balance` constructor on this line: `REGISTRY: Error: Balance: Negative number passed to unsigned type`

Reply:
Transfer on the parachain takes the planck and it must not have decimals. I guess the random things you see is because of the weak type checking of the DOT type that somehow converts this to an integer. On our side, we should make sure that it is a BN to ensure we don't have any decimal places.